Content.Server/GameObjects/Components/ExtinguisherCabinetComponent.cs

```diff
@@ -0,0 +1,115 @@
+
+        public override string Name => "ExtinguisherCabinet";
+
+        [ViewVariables] protected ContainerSlot ItemContainer;
+        private bool _opened = false;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            ItemContainer =
+                ContainerManagerComponent.Ensure<ContainerSlot>("extinguisher_cabinet", Owner, out _);
+        }
+
+        async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            if (ItemContainer.ContainedEntity != null || !eventArgs.Using.HasComponent<FireExtinguisherComponent>())
+            {
+                return false;
+            }
+            var handsComponent = eventArgs.User.GetComponent<IHandsComponent>();
+
+            if (!handsComponent.Drop(eventArgs.Using, ItemContainer))
+            {
+                return false;
+            }
+
+            UpdateVisuals();
+
+            return true;
+        }
+
+        bool IInteractHand.InteractHand(InteractHandEventArgs eventArgs)
+        {
+            if (_opened)
+            {
+                if (ItemContainer.ContainedEntity == null)
+                {
+                    _opened = false;
+                    ClickLatchSound();
+                }
+                else if (eventArgs.User.TryGetComponent(out HandsComponent hands))
+                {
+                    _notifyManager.PopupMessage(Owner.Transform.GridPosition, eventArgs.User,
+                        Loc.GetString("You take {0:extinguisherName} from the {1:cabinetName}", ItemContainer.ContainedEntity.Name, Owner.Name));
+                    hands.PutInHandOrDrop(ItemContainer.ContainedEntity.GetComponent<ItemComponent>());
+                }
+                else if (ItemContainer.Remove(ItemContainer.ContainedEntity))
+                {
+                    ItemContainer.ContainedEntity.Transform.GridPosition = Owner.Transform.GridPosition;
+                }
+            }
+            else
+            {
+                _opened = true;
+                ClickLatchSound();
+            }
+
+            UpdateVisuals();
+
+            return true;
+        }
+
+        void IActivate.Activate(ActivateEventArgs eventArgs)
+        {
+            _opened = !_opened;
+            ClickLatchSound();
+            UpdateVisuals();
+        }
+
+        private void UpdateVisuals()
+        {
+            if (Owner.TryGetComponent(out AppearanceComponent appearance))
+            {
+                appearance.SetData(ExtinguisherCabinetVisuals.IsOpen, _opened);
+                appearance.SetData(ExtinguisherCabinetVisuals.ContainsExtinguisher, ItemContainer.ContainedEntity != null);
+            }
+        }
+
+        private void ClickLatchSound()
+        {
+            EntitySystem.Get<AudioSystem>() // Don't have original click, this sounds close
+                .PlayFromEntity("/Audio/Machines/machine_switch.ogg", Owner, AudioHelpers.WithVariation(0.15f));
```

Comment:
This should be made into a property in ExposeData

Reply:
the .ogg file name? Expose data as in variable viewer thingy?

Reply:
https://github.com/space-wizards/space-station-14/blob/master/Content.Server/GameObjects/Components/Strap/StrapComponent.cs#L136

Reply:
done